app/models/user.rb

```diff
@@ -125,6 +126,19 @@ def set_avatar_from_twitter
   rescue
   end
 
+  # Extract the information that is available after OmniAuth authentication.
+  def self.from_omniauth(auth, params)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.provider = auth.provider
```

Comment:
When we remove that, HomeController#index can't find the user upon redirect to the homepage (see "User Load ... WHERE "users"."id" IS NULL")
![image](https://cloud.githubusercontent.com/assets/4050144/18214526/4302800c-7102-11e6-9b16-d7106c660e0d.png)

and we get the following UI issue:

![image](https://cloud.githubusercontent.com/assets/4050144/18214334/47051e36-7101-11e6-84de-a1b04de3dd99.png)

Any ideas where this is breaking down? Something in the devise signin?